parsons/openfield/openfield.py

```diff
@@ -0,0 +1,543 @@
+import json
+import logging
+import requests
+
+from parsons.etl.table import Table
+from parsons.utilities import check_env
+
+logger = logging.getLogger(__name__)
+
+
+class FailureException(Exception):
+    def __init__(self, resp, message="Error in request"):
+        self.status_code = resp.status_code
+        message = f"{resp.status_code} - {message}"
+
+        try:
+            self.json = resp.json()
+        except ValueError:
+            self.json = None
+            if resp.text:
+                message = f"{message}\n{resp.text}"
+
+        super(FailureException, self).__init__(message)
+
+
+class OpenField:
+    """
+    Instantiate the OpenField class
+
+    `Args:`
+        domain: str
+            The OpenField domain (e.g. ``org-name.openfield.ai``)
+            Not required if ``OPENFIELD_DOMAIN`` env variable set.
+        username: str
+            The authorized OpenField username.
+            Not required if ``OPENFIELD_USERNAME`` env variable set.
+        password: str
+            The authorized OpenField user password.
+            Not required if ``OPENFIELD_PASSWORD`` env variable set.
+    """
+
+    _default_headers = {
+        "content-type": "application/json",
+        "accepts": "application/json",
+    }
+
+    def __init__(self, domain=None, username=None, password=None):
```

Comment:
I'm surprised you didn't use the [API Connector class](https://github.com/move-coop/parsons/blob/main/parsons/utilities/api_connector.py) as the client here. Is that class not suitable for your purposes, or were you unaware it existed? If you chose not to use it, I'd love to get your feedback on what made it unworkable.

Reply:
Just unaware! I can refactor to use it as it seems better to have a standard in place like this now, for sure.